### PR TITLE
#1165: Parse more Open Graph metadata

### PR DESCRIPTION
**Pull request by joshuatbrown**
## Issues covered
#1165 

## Description
Parses more of the Open Graph metadata from the HTML, like the title, URLs, and type.

## How to test
Run the unit tests

## Notes
Skipping CHANGELOG since the previous entry covers it:
> Added SwiftSoup to parse Open Graph metadata. [#1165]

### CODE REVIEW

File: Nos/Models/OpenGraph/SoupOpenGraphParser.swift [hunk 1]
```diff
@@ -3,30 +3,114 @@ import SwiftSoup
...
-    ///   - document: The HTML document.
-    /// - Returns: The value of the Open Graph property, or `nil` if none is found.
-    private func openGraphProperty(_ property: OpenGraphProperty, from document: Document) -> String? {
-        try? document.select("meta[property=\(property.rawValue)]").attr("content")
+    /// Gets the Open Graph image URL from the given HTML document.
+    /// - Parameter document: The HTML document.
+    /// - Returns: The Open Graph image URL, or `nil` if none is found.
+    /// - Note: The image URL may be in a variety of properties, including `"og:image"`, `"og:image:url`, or
+    ///        `"og:image:secure_url"`.
+    func imageURL(from document: Document) -> String? {
```
Comment (mplorentz) [inline Nos/Models/OpenGraph/SoupOpenGraphParser.swift:88]: Aren't we going to want this and the following function to return `URL`? They don't seem that useful in `String` form, but maybe I am missing something. Anyway it's an easy refactor later.

**Review by mplorentz — APPROVED** [review]
Looks good! One suggestion, feel free to ignore.

File: Nos/Models/OpenGraph/SoupOpenGraphParser.swift (hunk 1 above)
Comment (joshuatbrown) [reply Nos/Models/OpenGraph/SoupOpenGraphParser.swift:88]: Yeah, we definitely are. I'll do it now.

Comment (joshuatbrown) [reply Nos/Models/OpenGraph/SoupOpenGraphParser.swift]: Done! 🚀

**Pull request merged**